routemaster/controllers/parser.rb

```
@@ -0,0 +1,38 @@
+require 'routemaster/controllers'
+require 'routemaster/mixins/log'
+require 'sinatra/base'
+
+module Routemaster
+  module Controllers
+    module Parser
+      module Helpers
+        include Mixins::Log
+
+        def data
+          @parsed_data
+        end
+      end
+
+      def self.registered(app)
+        app.helpers Helpers
+
+        app.set :parse do |format|
+          condition do
+            case format
+            when :json
+              begin
+                @parsed_data = JSON.parse(request.body.read)
+              rescue JSON::ParserError => e
+                _log.warn("JSON parse error: '#{e.message}'")
```

if we re-raise the error/send it to the exception tracker, will the increased level of context (such as the service submitting invalid JSON to Routemaster) be useful in this scenario?

---

Well, you'd get (some) context in RM's logs. Given this is a client error, it doesn't IMHO qualify as "RM is broken", so no exception should be raised.
The client should raise an exception the 400 it'll receive though.

Are you concerned about "missing" the fact there were bad events, for instance?
If so, we could add the #responses by HTTP status to our monitoring.

---

Yeah, I completely agree this case isn't exception-level of broken - I'm happy with monitoring HTTP response statuses and a papertrail alert on this error, does that sound okay?